src/storage/db_interface_frontend.py

```diff
@@ -24,22 +24,19 @@ def get_meta_list(self, firmware_list=None):
             firmware_list = self.firmwares.find()
         for firmware in firmware_list:
             if firmware:
-                if 'tags' in firmware:
-                    tags = firmware['tags']
-                else:
-                    tags = dict()
-                if firmware['processed_analysis']['unpacker']['file_system_flag']:
-                    unpacker = self.retrieve_analysis(deepcopy(firmware['processed_analysis']))['unpacker']['plugin_used']
-                else:
-                    unpacker = firmware['processed_analysis']['unpacker']['plugin_used']
-                tags[unpacker] = TagColor.LIGHT_BLUE
-                if 'submission_date' in firmware:
-                    submission_date = firmware['submission_date']
-                else:
-                    submission_date = 0
+                tags = firmware['tags'] if 'tags' in firmware else dict()
+                tags[self._get_unpacker_name(firmware)] = TagColor.LIGHT_BLUE
+                submission_date = firmware['submission_date'] if 'submission_date' in firmware else 0
                 list_of_firmware_data.append((firmware['_id'], self.get_hid(firmware['_id']), tags, submission_date))
         return list_of_firmware_data
 
+    def _get_unpacker_name(self, firmware):
+        if 'unpacker' not in firmware['processed_analysis']:
+            return 'NOP'
```

Comment:
why "NOP"? 
Is this a case that could happen regularly or should there be a log message?

Reply:
As stated above, this will happen regularly for firmware images with multiple layers.
I saw `NOP` as fitting, since this is the name of our unpack plugin that _runs_ whenever no extraction is wanted (e.g. blacklisted file types such as **.doc** or **.pdf**).
So as unpack depth is a special case of _should-not-be-extracted_ this looks sort of coherent.